src/mtev_http1.c

```diff
@@ -996,6 +1000,24 @@ mtev_http1_session_req_consume_read(mtev_http1_session_ctx *ctx,
       }
     }
     else {
+      /* Check if we have enough space in this bchain node for the chunk header "$chunk_size\r\n" */
+      if(head->allocd - head->start < MAX_CHUNK_HEADER_SIZE) {
+        mtevL(http_debug, "[fd=%d] Too little space in head node (%d). Reallocating.\n",
+              CTXFD(ctx), (int) (head->allocd - head->start));
+        struct bchain *new_head = ALLOC_BCHAIN(DEFAULT_BCHAINSIZE);
```

Comment:
I wonder if allocd is large enough, could we just memmove the start back to 0 in the same bchain without allocating a new one and freeing the old one.  The fact that we free the old one leads me to "yes".

Reply:
I think that's true. Let me fix this.

Reply:
Updated.